packages/data-grid/src/table/Table.jsx

```diff
@@ -138,6 +138,13 @@ class Table extends Component {
     return name;
   };
 
+  handleParentClick = (e, rowData, rowKey, rowIndex) => {
+    const { onRowClick } = this.props;
+    if (e.target.id !== `dg-checkbox-${rowIndex}`) {
```

Comment:
maybe just check for the classname here instead of adding ids to the elements. Checboxes that you wanna track have the same classname